rfcs/0131-Implementing Checks API in tc-github while preserving Statuses API.md

```diff
@@ -0,0 +1,63 @@
+# RFC 131 - Implementing Checks API
+* Comments: [#130](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/130)
+* Proposed by: @djmitche, @owlishDeveloper (author of this document)
+
+# Summary
+
+We want to implement Checks API without dropping the support os Statuses API. Statuses API would be the default. Our users
+will have the freedom to choose which API they want by indicating their choice in `.taskcluster.yml`.
+
+## Motivation
+
+So GitHub has come up with a new API to create status indicators in its UI, called Checks.
+Checks API isn't in any way compatible with the current Statuses API - status created with one cannot be updated with another;
+they are two totally separate sets of data. Calling both APIs would result in two sets of status indicators in the UI.
+Any third party tools and integrations that rely on Statuses API to work would stop working after we switched to Checks API.
+
+We want to implement Checks because it seems to be the future, and because it gives a much richer output and a whole bunch 
+of controls for our users.
+
+At the same time, we want to preserve compatibility with Statuses because some of our users use tools and integration that 
+rely on them.
+
+# Details
+
+### The general idea
+Currently we create a task and a status in a single function that listens
+to github webhooks. The idea is to create a task in separate function that listens to webhooks and creates tasks;
+then listen to the pulse message (namely, `task defined`) on a certain queue that signals about creation of a task.
+The messages would go to this or that queue based on some indication in `taskcluster.yml` - like a line `reporting: 'checks-v1'`
+in a task definition.
+
+### Details on the user interface
+The user interface in this case is `.taskcluster.yml`. The presence of the indication in the `.taskcluster.yml` would 
+result in adding a route (like a path to a queue?) to the task.
+We can also have something like `reporting: ['github-checks', 'github-statuses', 'irc://irc.mozilla.org:6667/#myproject']`
+to report through various channels. Status API will be enabled by default for all tasks; Checks API will be disabled
+for all tasks. Users will be able to set up either Statuses API or Checks API, as well as both for each individual task.
```

Comment:
Does GitHub endorse using both APIs concurrently?

Reply:
In `servo/servo` and other Servo-related repositories, is it somewhat typical to a have more than one system using Statuses independently. (Taskcluster-GitHub can be one of them.) Do you know of any issue with some but not all of those systems migrating to Checks, and still having all the information show up in a given PR at the same time?

Reply:
If those systems ask github for status of the PR using Statuses API only, and the status was created with Checks API, those systems will get, broadly speaking, `undefined` in return to their request. So they won't be working. Is that what you meant?